samples/matter/thermostat/src/temp_sensor_manager.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "temp_sensor_manager.h"
+#include "app_config.h"
+#include "app_task.h"
+#include "app_event.h"
+#include "temperature_measurement/real_temp.h"
+#include "temperature_measurement/mocked_temp.h"
+#include "binding_handler.h"
+
+#ifdef CONFIG_REAL_SENSOR
+#include "temperature_measurement/real_temp.h"
+#else
+#include "temperature_measurement/mocked_temp.h"
+#endif
+
+
+
+
+LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
+
+using namespace chip;
+using namespace chip::DeviceLayer;
+
+
+constexpr EndpointId kThermostatEndpoint = 1;
+
+
+k_timer sSensorTimer;
+
+CHIP_ERROR TempSensorManager::Init()
+{
+
+    int16_t initValue            = 0;
+    
+    PlatformMgr().LockChipStack();
+    app::Clusters::Thermostat::Attributes::LocalTemperature::Set(kThermostatEndpoint, initValue);
+    PlatformMgr().UnlockChipStack();
+
+    AppEvent event;
+    event.Handler            = TempSensorManager::SensorTimerEventHandler;
+    AppTask::Instance().PostEvent(event);
+
+    k_timer_init(&sSensorTimer, &TempSensorManager::TimerEventHandler, nullptr);
+    k_timer_user_data_set(&sSensorTimer, this);
+    k_timer_start(&sSensorTimer, K_MSEC(kSensorTimerPeriodMs), K_NO_WAIT);
```

Comment:
You start the timer here and in `SensorTimerEventHandler` method. Do you think it is correct? When the timer is initialized?